docs/user-guide/at-tls-configuration.md

```diff
@@ -1,23 +1,676 @@
+TTLSEnvironmentAction ZoweDCServerEnvironmentAction
+{
+  HandshakeRole Server
+  EnvironmentUserInstance 0
+  TTLSEnvironmentAdvancedParmsRef ServerEnvironmentAdvParms
+  TTLSKeyringParmsRef ZoweKeyring
+  TTLSSignatureParmsRef TNESigParms
+}
+
+# Keyring, used for TLS, will be used also to load trusted certificates
+TTLSKeyringParms ZoweKeyring
+{
+  Keyring ZWEKRNG
+}
+
+# Advanced TLS settings, choose TLS versions supported.
+TTLSEnvironmentAdvancedParms ServerEnvironmentAdvParms
+{
+  ClientAuthType Full # Support optional Client Certificate authentication
+  ApplicationControlled Off
+  Renegotiation Disabled
+  SSLv2 Off
+  SSLv3 Off
+  TLSv1 Off
+  TLSv1.1 Off
+  TLSv1.2 On
+  TLSv1.3 On
+}
+
+# Server Connection Action for API ML core services.
+TTLSConnectionAction ZoweServerConnectionAction
+{
+  HandshakeRole ServerWithClientAuth # API ML Core Services use Client Certificate authentication
+  TTLSCipherParmsRef CipherParms
+  TTLSConnectionAdvancedParmsRef ZoweConnectionAdvParms
+  TTLSSignatureParmsRef TNESigParms
+}
+
+# API ML Server connection action.
+# Certificate label indicates which certificate is used in the client certificate authentication process between core services.
+TTLSConnectionAdvancedParms ZoweConnectionAdvParms
+{
+  ApplicationControlled Off
+  ServerCertificateLabel apimlcert
+  CertificateLabel apimlcert
+  SecondaryMap Off
+}
+
+# Example outbound TTLS rule for a client calling API ML
+# In this scenario this client (a southbound service) presents client certificate to authenticate (for example during onboarding)
+TTLSRule ZoweClientRule
+{
+  LocalAddr All
+  LocalPortRange 1024-65535
+  RemoteAddr All
+  RemotePortRange 7553-7555 # API ML Core services ports
+  Jobname ZWE1*
+  Direction Outbound
+  TTLSGroupActionRef ClientGroupAction
+  TTLSEnvironmentActionRef ApimlClientEnvironmentAction
+  TTLSConnectionActionRef ApimlX509ClientConnAction
+}
+
+# Example outbound rule for connections from API ML Gateway to a southbound service running in port 40030 (during request routing)
+# Note ConnectionAction doesn't configure a client certificate.
+TTLSRule ApimlServiceClientRule
+{
+  LocalAddr All
+  LocalPortRange 1024-65535
+  RemoteAddr All
+  RemotePortRange 40030 # Service ports
+  Jobname ZWE1AG*
+  Direction Outbound
+  TTLSGroupActionRef ClientGroupAction
+  TTLSEnvironmentActionRef ApimlClientEnvironmentAction
+  TTLSConnectionActionRef ApimlNoX509ClientConnAction
+}
+
+# Optional. Can configure the outbound connection from Gateway to work with AT-TLS while connecting to z/OSMF.
+TTLSRule ApimlZosmfClientRule
+{
+  LocalAddr All
+  LocalPortRange 1024-65535
+  RemoteAddr All
+  RemotePortRange 449
+  Jobname ZWE1AG*
+  Direction Outbound
+  TTLSGroupActionRef ClientGroupAction
+  TTLSEnvironmentActionRef ApimlClientEnvironmentAction
+  TTLSConnectionActionRef ApimlNoX509ClientConnAction
+}
+
+# Example outbound rule from app server to gateway.
+TTLSRule ApimlZLUXClientRule
```

Comment:
There is missing an outbound rule GW > ZAAS (7558) if I'm correct.

Reply:
done, thanks